### PR TITLE
Feature/module 1

### LINKED ISSUE #1: Module 1: Basics. Node.js fundamentals theory
### Node.js Introduction. Modules. NPM

- [x] 1. What is Node.js and how it could be used?
- [x] 2. How Node.js versioning is defined?
- [x] 3. What types of modules are used/supported in Node.js?
- [x] 4. What is NPM and how to use it? What is package.json, its structure and purpose?
- [x] 5. What are NPM scripts and how to use them? What are package dependencies and what are their main types?

### Events and Event Loop. Async Development Introduction

- [x] 1. How does Node.js interact with asynchronous code?
- [x] 2. What is Node.js Event Loop?
- [x] 3. How does it work? What is EventEmitter and how does event system work in Node.js?
- [x] 4. What is a memory leak and what causes it?

### How to work with filesystem in Node.js?

- [x] 1. What are the main pros and cons of interacting with filesystem synchronously and asynchronously?
- [x] 2. What is a Buffer? How can it be used?
- [x] 3. What is a Stream? What are the benefits of using them while interacting with filesystem?

### PR DESCRIPTION
**Pull request by andrei-shelenhouski**
### All the tasks are fulfilled to the full extent (Task 1.1,Task 1.2,Task 1.3).

- [x] Task 1.1
- [x] Task 1.2
- [x] Task 1.3:

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -101,4 +101,4 @@ dist
 .dynamodb/
 
 # TernJS port file
-.tern-port
+.tern-port
```
Comment (kuzanatoliy) [inline .gitignore]: Empty line

File: hw1/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "NodeJS"
+  ],
+  "author": "Andrei Shelenhouski (andrei_shelenhouski@epam.com)",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/manuminsk/nodejs-gmp2020Q3/issues"
+  },
+  "homepage": "https://github.com/manuminsk/nodejs-gmp2020Q3#readme",
+  "dependencies": {
+    "csvtojson": "^2.0.10",
```
Comment (kuzanatoliy) [inline hw1/package.json]: Please use fixed versions in dependencies

File: hw1/task1/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  if (!input || !input.length) {
+    process.stdout.write('Process finished...');
+    process.exit(0);
+  }
+
+  process.stdout.write(
+    input
+      .split('')
+      .reverse()
+      .join('')
```
Comment (kuzanatoliy) [inline hw1/task1/index.js]: Would be nice to add '\n'

File: hw1/task2/index.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  files.forEach((file) => {
+    if (file && file.length) {
+      const fileName = file.split(".");
+      const fileType = fileName.pop();
+
+      fileType === "csv" &&
+        pipeline(
+          fs.createReadStream(`./csv/${file}`),
+          csv(),
```
Comment (kuzanatoliy) [inline hw1/task2/index.js:23]: It would be better if you added your transform stream

File: hw1/task2/index.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+// https://github.com/Keyang/node-csvtojson
```
Comment (kuzanatoliy) [inline hw1/task2/index.js:1]: Looks like right

**Comment by kuzanatoliy** [comment]
Please, use only one main folder.
And you can add example of csv file for second ticket

File: hw1/task2/index.js (hunk 4 above)
Comment (andrei-shelenhouski) [reply hw1/task2/index.js:23]: As we've discussed I'll investigate this later.

**Pull request merged**